network/gossip/libp2p/libp2pNode_test.go

```diff
@@ -342,9 +340,86 @@ func (l *LibP2PNodeTestSuite) TestCreateStreamTimeoutWithUnresponsiveNode() {
 	assert.Error(l.T(), err)
 }
 
+// TestCreateStreamIsConcurrent tests that CreateStream calls can be made concurrently such that one blocked call
+// does not block another concurrent call.
+func (l *LibP2PNodeTestSuite) TestCreateStreamIsConcurrent() {
+
+	// bump up the unicast timeout to a high value
+	unicastTimeout = time.Hour
+
+	// create two regular node
+	goodPeers, goodAddrs := l.CreateNodes(2, nil, false)
+	defer l.StopNodes(goodPeers)
+	require.Len(l.T(), goodAddrs, 2)
+
+	// create a silent node which never replies
+	listener, silentNodeAddress := newSilentNode(l.T())
```

Comment:
a silent node is a node which just accepts the TCP connection and does not send any response back